### PR TITLE
adding proxy ansible

### PR DESCRIPTION
**Pull request by akanso**
Adding the proxy Ansible impacts the following files:

- ansible/env-basics.yml: add the proxy role to the playbook      

- ansible/group_vars/all: adding the proxy related variables

- ansible/roles/proxy/tasks/main.yaml: is the role that: 1) builds the proxy container image 2) sets up the certs directory to be mounted to the proxy container 3) instantiates the proxy container with the proper parameters

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
I see some problems here.

It seems to install the proxy on localhost rather than the proxy-{{env_name}} inventory group.

It builds the proxy image rather than use the existing already-built image.

File: ansible/roles/proxy/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  - name: create envs directory
+    file: path={{CERTS}} state=directory mode=0755
+
+
+  - stat: path={{ACERTS}}/hjserver.pem
+    register: hjs_pem
+
+  - name: copy the api-proxy-openssl.cnf
+    copy: src=../ansible/roles/keygen-shard/files/api-proxy-openssl.cnf dest=/{{ACERTS}}/api-proxy-openssl.cnf
+    when: hjs_pem.stat.exists == False
```
Comment (MikeSpreitzer) [inline ansible/roles/proxy/tasks/main.yaml]: Has this been tested in both Ansible 1 and Ansible 2?  I think the boolean literals differ between the two.

The reliable and general way to do this is `when: not hjs_pem.stat.exists|bool`

**Comment by akanso** [comment]
@MikeSpreitzer yes this PR is still not finalized, it is mainly to make sure it passes Tommaso's tests, so I created the branch. 

To make it run on the proxy-VM we have a few more changes to make, like modifying the keygen to place the certs there, and copy some proxy files. 

Regarding the image, it is still not pushed in the registry, so we are building locally, but this should change and it should be places with the other images

You are right about the Ansible version, I only tested on 2.x but we should make it work on previous versions as well.

**Comment by MikeSpreitzer** [comment]
I still see `hjs_pem.stat.exists == False` on one line.
I thought the plan is that this playbook will not do the build of the image, not even by including another playbook.  The build should be done independently, with this playbook simply using the image.

**Comment by MikeSpreitzer** [comment]
Why does the `proxy_context` Ansible variable exist?  Is there ever a reason for anyone to set it to a different value than its default?

I thought we were going to put temporary files in `{{lookup('env', 'HOME')}}/tmp/` rather than somewhere in the local copy of the git repo.

File: ansible/roles/proxy/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+# Setting up the certs for the Proxy container
```
Comment (MikeSpreitzer) [inline ansible/roles/proxy/tasks/main.yaml]: This comment looks misplaced

**Pull request merged**